src/pkgcheck/checks/python.py

```diff
@@ -40,6 +43,26 @@ def short_desc(self):
                 f"{self.eclass} instead of {self.dep_type} on {self.dep}")
 
 
+class PythonSingleUseMismatch(base.Warning):
+    """Package has mismatched PYTHON_SINGLE & PYTHON_TARGETS flags."""
+
+    __slots__ = ("category", "package", "version", "flags", "single_flags")
+
+    threshold = base.versioned_feed
+
+    def __init__(self, pkg, flags, single_flags):
+        super().__init__()
+        self._store_cpv(pkg)
+        self.flags = flags
+        self.single_flags = single_flags
+
+    @property
+    def short_desc(self):
+        return (f"Python package has mismatched Python flags in IUSE: "
+                f"PYTHON_TARGETS={sorted(self.flags)} but "
```

Comment:
The flag sorting should be done during initialization and make them tuples since all result attributes should be immutable.